share/shutter/resources/system/plugins/perl/spwatermark/spwatermark

```diff
@@ -362,7 +362,7 @@ sub apply_effect {
 
 	my $command2 = "composite " . $tmpfilename . 
 		" -gravity " . $gravity_combo->get_active_text . 
-		" " . $filename . " " . $tmpfilename2;
```

Comment:
better not to use backtick at all, but to use some other ways to run subprocess. Or at least escape filesnames and such.

Reply:
Well, with a system call it resulted in an error which I couldn't understand (here is my question about it: https://stackoverflow.com/questions/68628648/the-and-comma-operators-in-a-perl-system-call).

What else would you propose instead of backticks?

Reply:
If I just revert your previous change, the `system` call works fine for me. What was the issue?

Reply:
It crashed when clicking on "apply" three years ago.

Reply:
can you show logs of this?

it works perfectly for me, if I revert your old commit

Reply:
Sure! But there is nothing visible in the logs:

```
### Launch the plugin:


fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)
Redundant argument in printf at /usr/bin/shutter line 6967.
	Shutter::App::fct_execute_plugin(undef, ARRAY(0x5c3e7a4b20e8)) called at /usr/bin/shutter line 10201
	Shutter::App::dlg_plugin("[51] - Auswahl_529.png") called at /usr/bin/shutter line 5841
	Shutter::App::fct_plugin(Gtk3::ImageMenuItem=HASH(0x5c3e79423d08)) called at /usr/lib/perl5/5.38/vendor_perl/Glib/Object/Introspection.pm line 67
	eval {...} called at /usr/lib/perl5/5.38/vendor_perl/Glib/Object/Introspection.pm line 67
	Glib::Object::Introspection::__ANON__(Shutter::App=HASH(0x5c3e78497c10)) called at /usr/bin/shutter line 11059


fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)

fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"


fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)

fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)

fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)


###Hit the refresh button:


fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)

fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)

fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)

fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)
WARNING: The convert command is deprecated in IMv7, use "magick" instead of "convert" or "magick convert"


fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)

fct_update_tray_menu was called by Wnck::Screen=HASH(0x5c3e784a4488)


### Plugin window disappears, the original screenshot stays untouched

```

Reply:
Three years ago I wrote that "The code exectution just stops at this system call and emits no output." -- no error output which could be visible in the logs either, as it looks like.

Reply:
Please try #702

Reply:
Still crashes as in #379, unfortunately. Note for later, perl-ipc-run3 as new dependency required if we want to pursue this variant.

Reply:
But is the log any better there why it crashes?

Reply:
Nope, it's still the same...

Reply:
well, I don't know how to debug this further for you.

you can merge this PR, of course, but don't be surprised when the next issue opened will be "crash if the name includes the quote character" or something like that

can you try to write a small script (without any shutter or even gtk) which just runs these 2 commands via backticks / via run3, and see what it says?